code/modules/vehicles/mecha/equipment/tools/work_tools.dm

```diff
@@ -277,28 +279,44 @@
 				internal_rcd.ui_interact(driver)
 			return TRUE
 
+/obj/item/mecha_parts/mecha_equipment/rcd/proc/cancel_action()
+	SIGNAL_HANDLER
+
+	mecha_moved = TRUE
+
+/obj/item/mecha_parts/mecha_equipment/rcd/do_after_checks(atom/target)
+	// Mecha moved during action
+	if(mecha_moved)
+		return FALSE
```

Comment:
I understand that there's no better way to do this until we get datumized do_afters but just for the record, god this stinks

Reply:
We work with what we got